app/src/pages/CollectionPage/CollectionPage.tsx

```diff
@@ -114,6 +115,19 @@ export const CollectionPage = (): React.ReactElement => {
     setIsRefreshClicked(true);
   };
 
+  React.useEffect((): void => {
+    if (shouldRefreshAllTokens) {
+      if (!account) {
+        return;
+      }
+      notdClient.updateCollectionToken(address, account.address).then((): void => {
```

Comment:
please dont test with this its very expensive. test with the updateCollection function and only switch to this one once you know everything works